extensions/kafka-eight-simpleConsumer/src/main/java/io/druid/firehose/kafka/KafkaEightSimpleConsumerFirehoseFactory.java

```
@@ -0,0 +1,312 @@
+import io.druid.data.input.FirehoseV2;
+import io.druid.data.input.InputRow;
+import io.druid.firehose.kafka.KafkaSimpleConsumer.BytesMessageWithOffset;
+
+public class KafkaEightSimpleConsumerFirehoseFactory implements
+    FirehoseFactoryV2<ByteBufferInputRowParser>
+{
+  private static final Logger log = new Logger(
+      KafkaEightSimpleConsumerFirehoseFactory.class
+  );
+
+  @JsonProperty
+  private final List<String> brokerList;
+
+  @JsonProperty
+  private final List<Integer> partitionIdList;
+
+  @JsonProperty
+  private final String clientId;
+
+  @JsonProperty
+  private final String feed;
+
+  @JsonProperty
+  private final int queueBufferLength;
+
+  @JsonProperty
+  private boolean earliest;
+
+  private final List<PartitionConsumerWorker> consumerWorkers = new CopyOnWriteArrayList<>();
+
+  @JsonCreator
+  public KafkaEightSimpleConsumerFirehoseFactory(
+      @JsonProperty("brokerList") List<String> brokerList,
+      @JsonProperty("partitionIdList") List<Integer> partitionIdList,
+      @JsonProperty("clientId") String clientId,
+      @JsonProperty("feed") String feed,
+      @JsonProperty("queueBufferLength") Integer queueBufferLength,
+      @JsonProperty("resetBehavior") String resetBehavior
+  )
+  {
+    this.brokerList = brokerList;
+    this.partitionIdList = partitionIdList;
+    this.clientId = clientId;
+    this.feed = feed;
+
+    this.queueBufferLength = queueBufferLength == null ? 20000 : queueBufferLength;
+    log.info("queueBufferLength loaded as[%s]", this.queueBufferLength);
+
+    this.earliest = "latest".equalsIgnoreCase(resetBehavior) ? false : true;
+    log.info("Default behavior of cosumer set to earliest? [%s]", this.earliest);
+  }
+
+  private Map<Integer, Long> loadOffsetFromPreviousMetaData(Object lastCommit)
+  {
+    Map<Integer, Long> offsetMap = Maps.newHashMap();
+    if (lastCommit == null) {
+      return offsetMap;
+    }
+    Map<Object, Object> lastCommitMap = (Map) lastCommit;
```

can we throw a meaningful exception of the lastCommit is not a map?